### PR TITLE
v2ray: 4.33.0 -> 4.34.0

### PR DESCRIPTION
**Pull request by servalcatty**
###### Motivation for this change
Bump version.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
  ```
  1 package built:
  v2ray
  ```
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/v2ray/default.nix [hunk 1]
```diff
@@ -50,8 +50,8 @@ let
     buildPhase = ''
       runHook preBuild
 
-      go build -o v2ray v2ray.com/core/main
-      go build -o v2ctl v2ray.com/core/infra/control/main
+      go build -v -p $NIX_BUILD_CORES -o v2ray ./main
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/v2ray/default.nix:53]: Can we add -ldflgas="-s -w"?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108805` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>v2ray</li>
</ul>
</details>

File: pkgs/tools/networking/v2ray/default.nix (hunk 1 above)
Comment (servalcatty) [reply pkgs/tools/networking/v2ray/default.nix:53]: @SuperSandro2000 
`-s` is for strip, but we already have default fixupPhase to strip all executables. 
What's `-w`? `man ld` doesn't contain it.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108805` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>v2ray</li>
</ul>
</details>

File: pkgs/tools/networking/v2ray/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/networking/v2ray/default.nix:53]: Please add it anyway explicitly to ldflags.

https://stackoverflow.com/questions/22267189/what-does-the-w-flag-mean-when-passed-in-via-the-ldflags-option-to-the-go-comman

Comment (oxalica) [reply pkgs/tools/networking/v2ray/default.nix]: Fixed.

Comment (servalcatty) [reply pkgs/tools/networking/v2ray/default.nix]: @SuperSandro2000 It do reduce the executable size. Thanks.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108805` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>v2ray</li>
</ul>
</details>

**Pull request merged**